modules/olm-dependencies-best-practices.adoc

```diff
@@ -0,0 +1,39 @@
+// Module included in the following assemblies:
+//
+// * operators/olm-understanding-dependency-resolution.adoc
+
+[id="olm-dependency-best-practices_{context}"]
+= Dependency best practices
+
+When specifying dependencies, there are best practices you should consider.
+
+Depend on APIs or a specific version range of Operators::
+Operators can add or remove APIs at any time; always specify an `olm.gvk` dependency on any APIs your Operators requires. The exception to this is if you are specifying `olm.packageVersion` constraints instead.
+
+Set a minimum version::
+The Kubernetes documentation on API changes describes what changes are allowed for Kubernetes-style Operators. These versioning conventions allow an Operator to update an API without bumping the API version, as long as the API is backwards-compatible.
+
+For Operator dependencies, this means that knowing the API version of a dependency might not be enough to ensure the dependent Operator works as intended.
+
+For example:
+
+* TestOperator v1.0 provides v1alpha1 API version of the MyObject resource.
```

Comment:
I'm not sure what is the intention of X.Y versioning given we should keep the version format to be semver-based X.Y.Z.

Reply:
Looks like the right of the examples in https://olm.operatorframework.io/docs/concepts/olm-architecture/dependency-resolution/ used `1.0.0` except for these two isntances of `1.0`, so I'll updated them to be `1.0.0` as well.